Add Terraform configuration files for shirt ordering agent quickstart

koverholt opened the pull request:
Use Terraform to provision a Dialogflow CX agent in Google Cloud based on [Create a Dialogflow CX agent](https://cloud.google.com/dialogflow/cx/docs/quick/build-agent) quickstart.

nicain commented on dialogflow-cx/shirt-ordering-agent/flows.tf:
```diff
@@ -0,0 +1,83 @@
+  # Use a REST API call (instead of Terraform modules) to modify messages and
+  # routes in the default start flow and since Dialogflow creates this default
+  # start flow automatically
+  provisioner "local-exec" {
```
If this works, great; If you have time, I would suggest stuffing this into a script and invoking it there. The security/linting was tricky to get this working but you can follow something like https://github.com/GoogleCloudPlatform/contact-center-ai-samples/blob/53ef6d95a3e1d803de18f5b8ce9312b0a61284cf/dialogflow-cx/vpc-sc-demo/components/reverse_proxy_server/main.tf#L248

koverholt replied:
I'm going to leave this as-is for the moment, I did try a few different approaches before landing on this one. If I moved the command to a script, then I would need to handle the handoff of the Terraform triggers/variables to the bash script as well, which adds more complexity. The nice thing with the current approach is that I can grab attributes from flows, intents, and pages and re-use them in both the provision request and the destroy-time provisioner.

nicain reviewed (changes requested):
Please add license headers to all files (copy-paste from some of the other files in the repo)

Also, take a look at https://github.com/GoogleCloudPlatform/contact-center-ai-samples/blob/main/dialogflow-cx/vpc-sc-demo/deploy/terraform/test_terraform_module.py; Its a bit complicated, but I am using `terraform graph` to assert dependency relationships between resources. Not something you necessarily need to do, but this would give you some unit-testing for your terraform files.

You might consider a test script that runs terraform init/terraform plan in a docker container, to validate the terraform module configurations (esp as pinned provider versions will be upgraded by dependabot)

koverholt commented:
Thanks for the review and help with linting/dev setup!

- Fixed the markdown issues
- Added license headers to each file
- Added terraform_required_providers

I'll look into the two test setups that you mentioned and see what I come up with. Thanks!

koverholt commented:
> You might consider a test script that runs terraform init/terraform plan in a docker container, to validate the terraform module configurations (esp as pinned provider versions will be upgraded by dependabot)

Thanks for the tip on this and the link to your`test_terraform_module.py`, that is a nice test setup.

I set up tests in this shirt-ordering-agent example in commit 8bc0cd3 that includes the same approach with a Dockerfile, test script, and Python check to make sure the dependencies are in order. And I added a GitHub action as well. 👍
